src/controller/proxy/local.go

```diff
@@ -0,0 +1,181 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package proxy
+
+import (
+	"context"
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/manifest/manifestlist"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/controller/blob"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/replication/adapter/harbor/base"
+	"github.com/goharbor/harbor/src/replication/model"
+	"io"
+	"sync"
+	"time"
+)
+
+var (
+	mu       sync.Mutex
+	inflight map[string]interface{} = make(map[string]interface{})
+)
+
+// LocalInterface defines operations related to local repo under proxy mode
+type LocalInterface interface {
```

Comment:
Do you envision the Local/Remote Interfaces be used by anything else by proxy controller?